vispy/gloo/buffer.py

```diff
@@ -191,7 +191,13 @@ def set_data(self, data, copy=False, **kwargs):
         """
         data = self._prepare_data(data, **kwargs)
         self._dtype = data.dtype
-        self._stride = data.strides[-1]
+        # This works around some strange NumPy bug where a float32 array
+        # of shape (155407, 1) was said to have strides
+        # (4, 9223372036854775807), which is crazy
+        if data.ndim == 2 and data.shape[1] == 1:
+            self._stride = data.strides[0]
+        else:
+            self._stride = data.strides[-1]
```

Comment:
Any idea if this bug still exists?

Reply:
Perhaps it was a bleeding-edge NumPy bug. I don't see it anymore so I've removed it